### DISCUSSION TITLE
error msg="Could not reset session during 1FA authentication for user 'user1'

### DISCUSSION THREAD

**Discussion by Alumni1506** [authelia/authelia#5795] [Q&A] [answered]
I had a functioning authelia/lldap setup which all of a sudden stopped working.
Each time i enter a valid username and password, authelia responds with an incorrect username/password error

Below is an example log obtained via portainer.

`time="2023-08-05T22:46:04Z" level=debug msg="Mark 1FA authentication attempt made by user 'user1'" method=POST path=/api/firstfactor remote_ip="redacted-client-ipaddr"
time="2023-08-05T22:46:04Z" level=debug msg="Successful 1FA authentication attempt made by user 'user1'" method=POST path=/api/firstfactor remote_ip="redacted-client-ipaddr"
time="2023-08-05T22:46:04Z" level=error msg="Could not reset session during 1FA authentication for user 'user1': NOAUTH Authentication required." method=POST path=/api/firstfactor remote_ip="redacted-client-ipaddr" stack="github.com/authelia/authelia/v4/internal/handlers/handler_firstfactor.go:80 FirstFactorPOST.func1\ngithub.com/authelia/authelia/v4/internal/middlewares/bridge.go:54           (*BridgeBuilder).Build.func1.1\ngithub.com/authelia/authelia/v4/internal/middlewares/headers.go:25          SecurityHeadersCSPNone.func1\ngithub.com/authelia/authelia/v4/internal/middlewares/headers.go:35          SecurityHeadersNoStore.func1\ngithub.com/authelia/authelia/v4/internal/middlewares/headers.go:16          SecurityHeaders.func1\ngithub.com/fasthttp/router@v1.4.14/router.go:414                            (*Router).Handler\ngithub.com/valyala/fasthttp@v1.43.0/http.go:154                             (*Response).StatusCode\ngithub.com/valyala/fasthttp@v1.43.0/server.go:2338                          (*Server).serveConn\ngithub.com/valyala/fasthttp@v1.43.0/workerpool.go:224                       (*workerPool).workerFunc\ngithub.com/valyala/fasthttp@v1.43.0/workerpool.go:196                       (*workerPool).getCh.func1\nruntime/asm_amd64.s:1594                                                    goexit"`

Can anyone advise what is going on before i nuke my setup?

**Comment by james-d-elliott** [comment]
Can you show some additional logs?  A single log line is rarely going to bring much clarity.

**Comment by james-d-elliott** [comment, marked as answer]
Oh actually, looks like a problem with the redis config. You need to configure the password that authelia uses.

**Reply by Alumni1506** [reply]
the config documentation mentions username and password are not required, so i left it blank.
I'll setup redis again with a username and password in the same way i did for nextcloud

**Reply by james-d-elliott** [reply]
The `NOAUTH Authentication required` error is from redis saying that it needs auth, it's only optional if you disable the redis password requirement.

**Reply by Alumni1506** [reply]
I disable the redis config and the initial issue went away. I just need to figure out why authelia is not talking to redis.

On another note, when I  complete a user authorisation, I get the following message

"Client authentication failed (e.g., unknown client, no client authentication included, or unsupported authentication method)."

What is above relating to?

**Reply by Alumni1506** [reply]
Resolution to my initial problem
- I essentially had to create a new redis container with a with name to get my setup working again.

Resolution to my second problem
- In nextcloud go to: Administraion > LDAP/AD integration > Expert Settings
-  Add uid to the "Internal Username" text box
-  Select "Clear Username-LDAP User Mapping"
-  In nextcloud's config.php file add
    - 'ldap_uid' => 'uid' as an entry under the 'oidc_login_attributes' => array (
    -  Initially I had set 'oidc_login_proxy_ldap' => true, which didn't work. Changing the setting to false worked for me.

**Reply by james-d-elliott** [reply]
That error generally means a mismatching client secret between authelia and the RP.
